src/xdp/trn_transit_xdp_hardware_offload.c

```diff
@@ -0,0 +1,510 @@
+	// ep types, they will have multiple attachments (e.g. LB endpoint).
+	if (ep->hosted_iface != -1) {
+		return XDP_PASS;
+	}
+
+	if (ep->eptype == TRAN_SCALED_EP) {
+		return XDP_PASS;
+	}
+
+	if (ep->nremote_ips == 0) {
+		return XDP_DROP;
+	}
+
+	trn_set_src_dst_ip_csum(pkt, pkt->ip->daddr, ep->remote_ips[0]);
+
+	return trn_rewrite_remote_mac(pkt);
+}
+
+static __inline int trn_process_inner_ip(struct transit_packet *pkt)
+{
+	pkt->inner_ip = (void *)pkt->inner_eth + pkt->inner_eth_off;
+	__u32 ipproto;
+
+	if (pkt->inner_ip + 1 > pkt->data_end) {
+		return XDP_ABORTED;
+	}
+
+	/* For whatever compiler reason, we need to perform reverse flow modification
+	 in this function instead of trn_switch_handle_pkt so we keep the orig_src_ip */
+	__u32 orig_src_ip = pkt->inner_ip->saddr;
+
+	pkt->inner_ipv4_tuple.saddr = pkt->inner_ip->saddr;
+	pkt->inner_ipv4_tuple.daddr = pkt->inner_ip->daddr;
+	pkt->inner_ipv4_tuple.protocol = pkt->inner_ip->protocol;
+	pkt->inner_ipv4_tuple.sport = 0;
+	pkt->inner_ipv4_tuple.dport = 0;
+
+	if (pkt->inner_ipv4_tuple.protocol == IPPROTO_TCP) {
+		pkt->inner_tcp = (void *)pkt->inner_ip + sizeof(*pkt->inner_ip);
+		if (pkt->inner_tcp + 1 > pkt->data_end) {
+			return XDP_ABORTED;
+		}
+
+		pkt->inner_ipv4_tuple.sport = pkt->inner_tcp->source;
+		pkt->inner_ipv4_tuple.dport = pkt->inner_tcp->dest;
+	}
+
+	if (pkt->inner_ipv4_tuple.protocol == IPPROTO_UDP) {
+		pkt->inner_udp = (void *)pkt->inner_ip + sizeof(*pkt->inner_ip);
+		if (pkt->inner_udp + 1 > pkt->data_end) {
+			return XDP_ABORTED;
+		}
+
+		pkt->inner_ipv4_tuple.sport = pkt->inner_udp->source;
+		pkt->inner_ipv4_tuple.dport = pkt->inner_udp->dest;
+	}
+
+	__be64 tunnel_id = trn_vni_to_tunnel_id(pkt->geneve->vni);
+
+	/* Lookup the source endpoint*/
+	struct endpoint_offload_t *src_ep;
+	struct endpoint_key_t src_epkey;
+	__builtin_memcpy(&src_epkey.tunip[0], &tunnel_id, sizeof(tunnel_id));
+	src_epkey.tunip[2] = pkt->inner_ip->saddr;
+	src_ep = bpf_map_lookup_elem(&endpoints_offload_map, &src_epkey);
+
+	/* If this is not the source endpoint's host,
+	skip reverse flow modification, or scaled endpoint modify handling */
+	if (pkt->scaled_ep_opt->type == TRN_GNV_SCALED_EP_OPT_TYPE &&
+	    pkt->scaled_ep_opt->scaled_ep_data.msg_type ==
+		    TRN_SCALED_EP_MODIFY &&
+	    src_ep && src_ep->hosted_iface != -1) {
+		return XDP_PASS;
+	}
+
+	/* Check if we need to apply a reverse flow update */
+	struct ipv4_tuple_t inner;
+	__builtin_memcpy(&inner, &pkt->inner_ipv4_tuple,
+			 sizeof(struct ipv4_tuple_t));
+
+	return trn_switch_handle_pkt(pkt, pkt->inner_ip->saddr,
+				     pkt->inner_ip->daddr, orig_src_ip);
+}
+
+static __inline int trn_process_inner_arp(struct transit_packet *pkt)
+{
+	unsigned char *sha;
+	unsigned char *tha = NULL;
+	struct endpoint_offload_t *ep;
+	struct endpoint_key_t epkey;
+	struct endpoint_offload_t *remote_ep;
+	__u32 *sip, *tip;
+	__u64 csum = 0;
+
+	pkt->inner_arp = (void *)pkt->inner_eth + sizeof(*pkt->inner_eth);
+
```

Comment:
please remove this blank line. Assignment and condition check are generally grouped together, I am aware that convention is not exactly followed in other places here.